i2c/bus_scan/bus_scan.c

```diff
@@ -33,13 +34,18 @@ bool reserved_addr(uint8_t addr) {
 int main() {
     // Enable UART so we can print status output
     stdio_init_all();
-
-    // This example will use I2C0 on GPIO4 (SDA) and GPIO5 (SCL)
-    i2c_init(i2c0, 100 * 1000);
-    gpio_set_function(4, GPIO_FUNC_I2C);
-    gpio_set_function(5, GPIO_FUNC_I2C);
-    gpio_pull_up(4);
-    gpio_pull_up(5);
+#if !defined(i2c_default) || !defined(PICO_DEFAULT_I2C_SDA_PIN) || !defined(PICO_DEFAULT_I2C_SCL_PIN)
+#warning i2c/bus_scane example requires a board with I2C pins
+    puts("Default I2C pins were not defined");
```

Comment:
Just out of curiosity - why the usage of `puts` here, when `printf` is used everywhere else?

Reply:
moon phase